y2020/ex17/src/lib.rs

```diff
@@ -12,27 +12,71 @@ struct Point<const D: usize>([i32; D]);
 #[derive(Debug)]
 struct Universe<const D: usize>(HashSet<Point<D>>);
 
-fn neighbours_of_point<const D: usize>(point: &Point<D>) -> Vec<Point<D>> {
-    let mut points = vec![];
-
-    for i in 0..3_i32.pow(D as u32) {
-        let mut relative_coords = [0; D];
-        let mut n = i;
-        for j in 0..D {
-            relative_coords[j] = (n % 3) - 1;
-            n /= 3;
+#[derive(Debug)]
+struct Game<const D: usize> {
+    universe: Universe<D>,
+    relative_neighbours: Vec<[i32; D]>,
```

Comment:
Is this a vector of "Points" (Array of D i32 elements) ?

Reply:
a vector of relative points: `[[-1,-1,-1], [-1,-1,0], ...]`.

I kept it simple and i am only storing the coordinates as `[i32;D]` here.